docs/resources/grant.md

```diff
@@ -0,0 +1,329 @@
+---
```

Comment:
@martin-walsh since this page is mostly duplicated from grants.md, we should collapse the 2 pages into one grants.md, and highlight the different usage of `databricks_grant` and `databricks_grants`

Reply:
Imho, two pages may help users to understand differences. I think that to avoid duplication we can just have links from another page to common things, like, syntax, etc.

Reply:
Do you guys need anything more from me here? I'm not really sure how you guys want your documentation structured.

Reply:
we should put a link on this page for the content that are the same as `databricks_grants.md`, e.g. line 27 onwards. Something like "See [databricks_grant.md] for more usage details

I've highlighted the part that should be called out differently to `databricks_grants`

Reply:
I can't see the highlight. Do you mean remove everything on `databricks_grants.md` from line 27 onwards ? If so, if it were me I would leave it. It is very user friendly to be able to copy valid syntax.